staging/src/k8s.io/cloud-provider/fake/fake.go

```diff
@@ -200,6 +201,7 @@ func (f *Cloud) GetLoadBalancerName(ctx context.Context, clusterName string, ser
 // EnsureLoadBalancer is a test-spy implementation of LoadBalancer.EnsureLoadBalancer.
 // It adds an entry "create" into the internal method call record.
 func (f *Cloud) EnsureLoadBalancer(ctx context.Context, clusterName string, service *v1.Service, nodes []*v1.Node) (*v1.LoadBalancerStatus, error) {
+	f.markUpdateCall(service, nodes)
```

Comment:
this was not called before

Reply:
I added this because the service sync path ends up calling `EnsureLoadBalancer` and I need the call to get registered so that I can validate this in the tests

Reply:
yeah, but we can not modify a general mock to fit into our custom logic in the test, if you want to add custom logic in the Ensure method just add an Ensure callback so people can override it

Reply:
> yeah, but we can not modify a general mock to fit into our custom logic in the test

At some point someone modified the general mock to fit the custom logic of these tests...why can't I not do that now?

Reply:
> At some point someone modified the general mock to fit the custom logic of these tests.

the problem is not customize the mock , the problem is to add a the updateCallback on the Ensure method, just add an enureCallback so it is correct  ... test code is the same as production code

Reply:
OK, I've updated the PR and added explicit field for the ensure calls (so both `EnsureCalls`/ `EnsureCallCb`)